README.md

```diff
@@ -1,57 +1,253 @@
+
+  <label>Post Description</label><br>
+  <textarea id="post_description" name="post[description]"></textarea><br>
+
+  <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
+  <input type="submit" value="Submit Post">
+</form>
+```
+
+If you refresh the page you will see that not only is the error fixed, but the elements are now also being printed out on the page! Running the specs you will see that our spec is now passing, so our form is working, however this might be one of the ugliest Rails forms I've ever seen, so let's do some re-factoring.
+
+
+## Using form helpers
+
+The ActionView has a number of methods we can use to streamline our form. Let's start by integrating a Rails form_tag element:
+
+```ERB
+<%= form_tag posts_path do %>
+  <label>Post title:</label><br>
+  <input type="text" id="post_title" name="post[title]"><br>
+
+  <label>Post Description</label><br>
+  <textarea id="post_description" name="post[description]"></textarea><br>
+
+  <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
+  <input type="submit" value="Submit Post">
+<% end %>
+```
+
+Running the tests you will see that all of the tests are still passing. If you go and look at the HTML that this generates you will see the following:
+
+```html
+<form action="/posts" accept-charset="UTF-8" method="post"><input name="utf8" type="hidden" value="&#x2713;" /><input type="hidden" name="authenticity_token" value="7B5m/x8roxG6bc1vbRxRhr/z+ql6D261+j6LwHeGjeial7hxyW+5zk6CTGLpY1aKUj9hzLPdtRhWfcX5i047Iw==" />
+  <label>Post title:</label><br>
+  <input type="text" id="post_title" name="post[title]"><br>
+
+  <label>Post Description</label><br>
+  <textarea id="post_description" name="post[description]"></textarea><br>
+
+  <input type="hidden" name="authenticity_token" id="authenticity_token" value="JC6cMiNK9k8nuWJdKp/3E29C/bALDkK9N6it3TeFEb1Sp0K89Q7skNNW41Cu4PAfgo5m1cLcmRCb6+Pky02ndg==" />
+  <input type="submit" value="Submit Post">
+</form>
+```
+
+The ```form_tag``` Rails helper is smart enough to know that we want to pass the form params using the POST method and it automatically render the HTML that we were writing by hand before.
+
+Now let's integrate some other form helpers to let Rails generate the input elements for us, for this form we'll be using the ```text_field_tag``` and ```text_area_tag``` tag and pass them the attributes with symbols. We'll also replace the HTML tag for the submit button with the ```submit_tag```:
+
+```ERB
+<%= form_tag posts_path do %>
+  <label>Post title:</label><br>
+  <%= text_field_tag :title %><br>
+
+  <label>Post Description</label><br>
+  <%= text_area_tag :description %><br>
+
+  <%= hidden_field_tag :authenticity_token, form_authenticity_token %>
+  
+  <%= submit_tag "Submit Post" %>
+<% end %>
+```
+
+This is all working on the page, however it broke some of our tests since it streamlined the ID attribute in the form, so let's update our spec:
```

Comment:
Show them the difference in the HTML from what the form tag had, to what the previous raw html had